### PR TITLE
control/controlclient: refactor in prep for optimized delta handling

### PR DESCRIPTION
**Pull request by bradfitz**
See issue. This is a baby step towards passing through deltas
end-to-end from node to control back to node and down to the various
engine subsystems, not computing diffs from two full netmaps at
various levels. This will then let us support larger netmaps without
burning CPU.

But this change itself changes no behavior. It just changes a func
type to an interface with one method. That paves the way for future
changes to then add new NetmapUpdater methods that do more
fine-grained work than updating the whole world.

Updates #1909

### CODE REVIEW

File: control/controlclient/direct.go [hunk 1]
```diff
@@ -1135,7 +1152,7 @@ func (c *Direct) sendMapRequest(ctx context.Context, isStreaming bool, cb func(*
 		c.expiry = &nm.Expiry
 		c.mu.Unlock()
 
-		cb(nm)
+		nu.UpdateFullNetmap(nm)
```
Comment (bradfitz) [inline control/controlclient/direct.go:1155]: @maisem, this is the core of the change: changing a func callback to an interface method call.

File: control/controlclient/auto.go [hunk 2]
```diff
@@ -480,11 +480,42 @@ func (c *Auto) unpausedChanLocked() <-chan struct{} {
...
+	stillAuthed := c.loggedIn
+	c.logf("[v1] mapRoutine: netmap received: %s", c.state)
+	c.mu.Unlock()
+
+	if stillAuthed {
+		c.sendStatus("mapRoutine-got-netmap", nil, "", nm)
+	}
+	// Reset the backoff timer if we got a netmap.
+	mrs.bo.BackOff(ctx, nil)
+}
```
Comment (bradfitz) [inline control/controlclient/auto.go:509]: This is purely movement, out of the func below.

It was closing over `backoff`, though, hence the new type.

File: control/controlclient/auto.go [hunk 3]
```diff
@@ -567,10 +580,10 @@ func (c *Auto) mapRoutine() {
 
 			if err != nil {
```
Comment (maisem) [inline control/controlclient/auto.go:581]: Err is always non-nil at this point.

**Review by maisem — APPROVED** [review]

File: control/controlclient/auto.go (hunk 3 above)
Comment (bradfitz) [reply control/controlclient/auto.go:581]: And it was before too-- glad we can actually see that easily now! 😂

**Pull request merged**